src/Controls/tests/DeviceTests/Elements/Entry/EntryTests.cs

```diff
@@ -52,6 +54,50 @@ public async Task MaxLengthIsReadOnlyValueTest()
 				Assert.True(platformControl.IsReadOnly);
 			});
 		}
+
+		
+		[Fact(DisplayName = "Unfocus will work when page is shown a 2nd time")]
+		public async Task UnFocusOnEntryAfterPagePop()
+		{
+			int unfocused = 0;
+			EnsureHandlerCreated(builder =>
+			{
+				builder.ConfigureMauiHandlers(handlers =>
+				{
+					handlers.AddHandler(typeof(Toolbar), typeof(ToolbarHandler));
+					handlers.AddHandler(typeof(NavigationPage), typeof(NavigationViewHandler));
+					handlers.AddHandler<Page, PageHandler>();
+					handlers.AddHandler(typeof(Window), typeof(WindowHandlerStub));
+					handlers.AddHandler(typeof(Entry), typeof(EntryHandler));
+
+				});
+			});
+
+			var entry = new Entry();
+			entry.Unfocused += (s, e) =>
+			{
+				if(!e.IsFocused)
+				{
+					unfocused++;
+				}
+
+			};
+			var navPage = new NavigationPage(new ContentPage { Content  = entry });
+			var window = new Window(navPage);
+
+			await CreateHandlerAndAddToWindow<WindowHandlerStub>(window, async (handler) =>
+			{
+				entry.Focus();
+				await Task.Delay(500);
```

Comment:
Why are the deals needed? Is there an event we can listen to know that the focus was performed. Using timers in tests will result in flaky test. You could move to listen an even and use an AutoResetEvent in the callback: https://learn.microsoft.com/en-us/dotnet/api/system.threading.autoresetevent?view=net-7.0

We do that a low in multithreaded tests in the iOS tests.

Reply:
Thanks, changed the test to use AutoResetEvent